src/julia.h

```diff
@@ -1568,9 +1594,9 @@ JL_DLLEXPORT void jl_exception_clear(void) JL_NOTSAFEPOINT;
 #define JL_NARGSV(fname, min)                           \
     if (nargs < min) jl_too_few_args(#fname, min);
 
-#define JL_TYPECHK(fname, type, v)                                      \
-    if (!jl_is_##type(v)) {                                             \
-        jl_type_error(#fname, (jl_value_t*)jl_##type##_type, (v));      \
+#define JL_TYPECHK(fname, type, v)                                              \
+    if (!jl_is_##type(v)) {                                                     \
+        jl_type_error(#fname, (jl_value_t*)jl_##type##_type, (jl_value_t*)(v)); \
```

Comment:
I'm not sure we should do this, though if we do, probably should change `JL_TYPECHKS` to be consistent.